Jinke

jink-e opened the pull request:
Add file chooser to soundTrack

Commit:
Merge branch 'master' into jinke

Commit:
Merge remote-tracking branch 'refs/remotes/origin/master' into jinke

Commit:
ignore #*

Commit:
Merge branch 'bryce_branch' into jinke

Commit:
Merge branch 'master' into jinke

Commit:
update from master

Commit:
new ignore of #*

Commit:
Merge branch 'master' into jinke

Commit:
new ignore in bin/

Commit:
new ignore file in /src

Commit:
ignore #* under src/

Commit:
reset back to master

Commit:
Merge branch 'master' into jinke

# Conflicts:
#	SeniorSlideshowProject/bin/.gitignore
#	SeniorSlideshowProject/bin/SlideshowMaker$1.class
#	SeniorSlideshowProject/bin/SlideshowMaker$2.class
#	SeniorSlideshowProject/bin/SlideshowMaker.class
#	SeniorSlideshowProject/bin/SoundTrack$1.class
#	SeniorSlideshowProject/bin/SoundTrack$Credits.class
#	SeniorSlideshowProject/bin/SoundTrack$JukeControls.class
#	SeniorSlideshowProject/bin/SoundTrack$JukeTable$1.class
#	SeniorSlideshowProject/bin/SoundTrack$JukeTable.class
#	SeniorSlideshowProject/bin/SoundTrack$Loading.class
#	SeniorSlideshowProject/bin/SoundTrack$PlaybackMonitor.class
#	SeniorSlideshowProject/bin/SoundTrack.class
#	SeniorSlideshowProject/src/.gitignore

Commit:
merge from master

Commit:
Add file chooser to sound track

Commit:
Merge branch 'master' into jinke

Commit:
add audioState when add audio file from file chooser

Commit:
merge #19 from master

Commit:
remove .class file under bin/Slides/

bpj0001 committed:
Update .gitignore

bpj0001 committed:
Update .gitignore

bpj0001 commented on SeniorSlideshowProject/src/AddAudioListener.java:
```diff
@@ -0,0 +1,42 @@
+			String selectedFilePath = fc.getSelectedFile().getPath();
+			if (selectedFilePath.contains(".jpg")){
+				ImageIcon icon = new ImageIcon(selectedFilePath);
+				SlideState mySlide = new SlideState(icon);
```
Why would you have an addAudioListener that is adding slides to the state machine instead of audio files? This doesn't make sense to me.

bpj0001 replied:
In fact I don't see you using this class anywhere, is this necessary?

bpj0001 commented on SeniorSlideshowProject/src/Slides/AudioState.java:
```diff
@@ -1,31 +1,58 @@
-	private double audioTime;
-	
-	public AudioState(String name)
+	public AudioState(AudioInputStream inputStream, String name)
```
You can remove the String from the constructor and get rid of the fileName member variable. We just want the inputStream that you will serialize.

bpj0001 commented on SeniorSlideshowProject/src/SlideshowMaker.java:
```diff
@@ -26,6 +26,8 @@
 
 	private static final long serialVersionUID = 1L;
+	
+	private SlideShowStateMachine stateMachine;
```
For some reason your PR has an outdated version of this file. There is already a state machine in the newest version. I'd try to re-update from master. Or maybe close this PR and open a new one if you already have the laster version of the file in there. I'm hesitant to say you should commit it if it overwrites the work already done.

bpj0001 reviewed (changes requested):
Overall the changes to your files are okay. However I've added some comments in the code area for addAudioListener and SlideShowMaker.java. There are some issues that need to be addressed. Fix those and I'll approve.

Commit:
Merge branch 'jinke' of https://github.com/WilliamParsons/SlideShow into jinke

Commit:
merge #19

Commit:
modify .gitignore

# Conflicts:
#	SeniorSlideshowProject/bin/.gitignore
#	SeniorSlideshowProject/bin/SlideshowMaker$1.class
#	SeniorSlideshowProject/bin/SlideshowMaker$2.class
#	SeniorSlideshowProject/bin/SlideshowMaker.class

Commit:
comment on variable fileName in AudioState

Commit:
Merge branch 'master' into jinke
